xapian-letor/ranker/ranker.cc

```diff
@@ -192,35 +192,31 @@ write_to_file(const std::vector<Xapian::FeatureVector> & list_fvecs, const strin
     }
 }
 
+// Query file is in the format: <qid> '<query_string>'
 static std::pair<string, string>
 parse_query_string(const string & query_line, int line_number)
 {
-    vector<string> token;
-    size_t i = 0;
-    size_t j = query_line.find_first_of(' ', i);
-    token.push_back(query_line.substr(i, j - i));
+    string::size_type i = 0;
+    string::size_type j = query_line.find_first_of(' ', i);
+    string qid = query_line.substr(i, j - i);
+    string querystr;
     if (j != string::npos) {
-	i = query_line.find_first_of("'", j);
-	j = string::npos;
-	if (j > i) {
-	    token.push_back(query_line.substr(i, j - i));
+	i = query_line.find_first_not_of("' ", j);
+	j = query_line.length() - 1;
+	// check if the last character is '
+	if (query_line[j] != '\'') {
```

Comment:
We need to take care to handle the case where there's a `'` and nothing after that appropriately:

```
1234 '
```

The closing quote must not be at the same string index we interpreted as being the opening quote.

Reply:
fixed in a fixup commit.